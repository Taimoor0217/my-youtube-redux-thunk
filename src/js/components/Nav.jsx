import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Search from './Search';
import { getVideos } from '../actions';

const Nav = (props) => {
  return (
    <div className="nav">
      <Search {...props} />
    </div>
  )
}

Nav.propTypes = {
  getVideos: PropTypes.func.isRequired,
}

export default connect(null, {getVideos})(Nav);