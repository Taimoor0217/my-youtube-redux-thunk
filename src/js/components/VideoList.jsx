import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import VideoListEntry from './VideoListEntry';
import { selectVideo } from '../actions';

class VideoList extends Component {

  createList() {
    let { videoList, selectVideo } = this.props;
    return videoList.map((item, i) => {
      return <VideoListEntry key={i} video={item} handleSelectedVideo={selectVideo} />
    })
  }

  render() {
    let { videoList, isLoading, error } = this.props;
    return (
      <div className="videoList">
        {
          isLoading ? <div>Loding Videos....</div> :
          error ? <div>{error.message}</div> : this.createList()
        }
      </div>
    );
  }
}

VideoList.propTypes = {
  videoList: PropTypes.array.isRequired,
  selectVideo: PropTypes.func.isRequired,
}

function mapStateToProps({videoList}) {
  return {
    videoList: videoList.videos,
    isLoading: videoList.isLoading,
    error: videoList.error
  }
};

export default connect(mapStateToProps, {selectVideo})(VideoList);