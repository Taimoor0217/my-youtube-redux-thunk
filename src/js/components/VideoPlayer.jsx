import React, { Component } from 'react';
import PropTypes from 'prop-types';


const VideoPlayer = ({ seletedVideo }) => {
  if(Object.keys(seletedVideo).length) {
    var seletedVideoURL = "https://www.youtube.com/embed/" + seletedVideo.id.videoId
  }
  return (
    <div className="videoPlayer">
      <div className="player embed-responsive embed-responsive-16by9">
        <iframe 
          className="embed-responsive-item"
          src={selectedVideoURL} 
          frameborder="0"
          allowFullScreen
        />
      </div>
      <div className="video-details">
        <h4 className="video-title">
          {Object.keys(selectedVideo).length ? selectedVideo.snippet.title : ""}
        </h4>
        <p className="video-descriptions">
          {Object.keys(seletedVideo).length ? selectedVideo.snippet.description : ""}
        </p>
      </div>
    </div>
  )
};

VideoPlayer.propTypes = {
}

export default VideoPlayer;