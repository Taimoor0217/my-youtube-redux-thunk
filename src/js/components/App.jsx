import React, { Component } from 'react';
import { connect } from 'react-redux';

import Nav from './Nav';
import VideoPlayer from './VideoPlayer';
import VideoList from './VideoList';
import { getVideos } from '../actions';

class App extends Component {
  render() {
    return (
      <div id="app">
        <Nav />
        <div className="col-md-7">
          <VideoPlayer />
        </div>
        <div className="col-md-5">
          <VideoList />
        </div>
      </div>
    );
  }

  componentDidMount() {
    const { getVideos } = this.props;
    getVideos('javascript', true);
  }
}

export default connect(null, {getVideos})(App);