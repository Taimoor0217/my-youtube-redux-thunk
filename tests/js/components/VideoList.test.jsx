import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { VideoList } from '../../../src/js/components/VideoList';

Enzyme.configure({ adapter: new Adapter() });

describe('VideoList Component', () => {
  let videoList, selectVideo;

  beforeEach(() => {
    videoList = [
      {
        id: { videoId: '1' },
        snippet: {
          title: 'some title 1',
          description: 'some description 1',
          thumbnails: {
            default: {
              url: 'someurl 1'
            }
          }
        }
      },
      {
        id: { videoId: '2' },
        snippet: {
          title: 'some title 2',
          description: 'some description 2',
          thumbnails: {
            default: {
              url: 'someurl 2'
            }
          }
        }
      }
    ],
    selectVideo = jest.fn();
  });

  it('renders', () => {
    let wrapper = shallow(
      <VideoList 
        selectVideo={selectVideo}
        videoList={videoList}
        isLoading={false}
        error={null}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Loading message', () => {
    let wrapper = shallow(
      <VideoList 
        selectVideo={selectVideo}
        videoList={videoList}
        isLoading={true}
        error={null}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Error message', () => {
    let wrapper = shallow(
      <VideoList 
        selectVideo={selectVideo}
        videoList={videoList}
        isLoading={false}
        error={{ message: 'Error while loading videos' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  
  it('passes on the selectVideo function', () => {
    expect(selectVideo.mock.calls.length).toBe(0);
  });

})