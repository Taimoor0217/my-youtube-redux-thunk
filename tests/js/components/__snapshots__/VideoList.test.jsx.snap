// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoList Component renders 1`] = `
<div
  className="videoList"
>
  <VideoListEntry
    handleSelectedVideo={[MockFunction]}
    key="1"
    video={
      Object {
        "id": Object {
          "videoId": "1",
        },
        "snippet": Object {
          "description": "some description 1",
          "thumbnails": Object {
            "default": Object {
              "url": "someurl 1",
            },
          },
          "title": "some title 1",
        },
      }
    }
  />
  <VideoListEntry
    handleSelectedVideo={[MockFunction]}
    key="2"
    video={
      Object {
        "id": Object {
          "videoId": "2",
        },
        "snippet": Object {
          "description": "some description 2",
          "thumbnails": Object {
            "default": Object {
              "url": "someurl 2",
            },
          },
          "title": "some title 2",
        },
      }
    }
  />
</div>
`;

exports[`VideoList Component renders Error message 1`] = `
<div
  className="videoList"
>
  <div>
    Error while loading videos
  </div>
</div>
`;

exports[`VideoList Component renders Loading message 1`] = `
<div
  className="videoList"
>
  <div>
    Loading Videos...
  </div>
</div>
`;
